Instant voice demo improvements - part 02

filipi87 opened the pull request:
#### Please describe the changes in your PR. If it is addressing an issue, please reference that as well.

aconchillo commented on src/pipecat/processors/frameworks/rtvi.py:
```diff
@@ -844,6 +849,12 @@ def __init__(
+        if self._transport:
+            input_transport = self._transport.input()
+            if isinstance(input_transport, BaseInputTransport):
+                input_transport.enable_audio_in_stream_on_start(False)
```
I think we can just do:

```
        if self._transport:
            self._transport.input().enable_audio_in_stream_on_start(False)
```

`input()` should always be a `BaseInputTransport`.

aconchillo replied:
or probably better, to make sure we have an input transport:

```
        if self._transport and self._transport.input():
            self._transport.input().enable_audio_in_stream_on_start(False)
```

aconchillo replied:
One more though... Since we are using the input transport everywhere, maybe:

```
self._input_transport = transport.input() if transport else None

if self._input_transport:
            self._input_transport.enable_audio_in_stream_on_start(False)
```

filipi87 replied:
I am going to implement your suggestion of creating a variable for `_input_transport`, so we can use it anywhere. That will already help a lot.

filipi87 replied:
Done!

aconchillo commented on src/pipecat/processors/frameworks/rtvi.py:
```diff
@@ -1025,9 +1038,40 @@ async def _handle_message(self, message: RTVIMessage):
             logger.warning(f"Exception processing message: {e}")
 
     async def _handle_client_ready(self, request_id: str):
+        logger.info("_handle_client_ready")
```
This should probably be `debug` and something like `Received client-ready` or something like that.

filipi87 replied:
Done!

aconchillo commented on src/pipecat/processors/frameworks/rtvi.py:
```diff
@@ -1025,9 +1038,40 @@ async def _handle_message(self, message: RTVIMessage):
+
+        input_transport = self._transport.input()
+        if not isinstance(input_transport, BaseInputTransport):
+            return
```
Since we store `_input_transport`:

```
        if not self._input_transport:
            return
```

filipi87 replied:
Done!

aconchillo commented on src/pipecat/transports/services/daily.py:
```diff
@@ -839,6 +838,13 @@ def __init__(self, client: DailyTransportClient, params: DailyParams, **kwargs):
         return self._vad_analyzer
 
+    def start_audio_in_streaming(self):
+        logger.debug(f"Start receiving audio")
```
Should we move the log line inside the `if` below in case audio in is disabled?

filipi87 replied:
Done!

aconchillo commented:
Cool! I think it makes more sense like this. Looks good! Just added minor comments.

filipi87 commented:
> Cool! I think it makes more sense like this. Looks good! Just added minor comments.
Yep, I agree. It makes more sense handling the messages inside RTVI.

I believe I have addressed all the comments. It should be ready for another round of review.

aconchillo commented:
LGTM! It would be nice to rebase and cleanup the commits to just a few commits that make sense.
